### DISCUSSION TITLE
Problem/Bug using dot notation for numbered objects

### DISCUSSION THREAD

**Discussion by wuifdesign** [react-hook-form/react-hook-form#2911] [Q&A]
Hi,

I have a problem and don't know if it is a bug. if have an object an want to update it with the form. but the output becomes an array.

```
var valuesIn = {
  test: {
    '123': 'this is a test',
  }
}

<input type="text" name={`test.123`} ref={register()}/>

getValues() ->  {
  test: [empty × 122, 'this is a test'],
}
```

is it possible to force an object even when using strings that look like numbers as keys?

**Comment by bluebill1049** [comment]
<img width="941" alt="Screen Shot 2020-09-15 at 6 23 58 pm" src="https://user-images.githubusercontent.com/10513364/93185350-a12a4d00-f780-11ea-9db9-b7f9e1b03ecf.png">

it's a limitation.

**Reply by wuifdesign** [reply]
oh, i see. can't get past it? or any workaround to get the right values?

**Reply by bluebill1049** [reply]
too hard with regex to get it working with `number`
